x-pack/plugins/apm/public/components/shared/TransactionActionMenu/TransactionActionMenu.tsx

```diff
@@ -43,168 +49,189 @@ function ActionMenuButton({ onClick }: { onClick: () => void }) {
+        'xpack.apm.transactionActionMenu.showHostLogsLinkLabel',
+        { defaultMessage: 'Show host logs' }
+      ),
+      condition: hostName,
+      path: `/link-to/host-logs/${hostName}`,
+      query: { time }
+    },
+    {
+      icon: 'loggingApp',
+      label: i18n.translate(
+        'xpack.apm.transactionActionMenu.showTraceLogsLinkLabel',
+        { defaultMessage: 'Show trace logs' }
+      ),
+      target: traceId,
+      hash: `/link-to/logs`,
+      query: { time, filter: `trace.id:${traceId}` }
+    },
+    {
+      icon: 'infraApp',
+      label: i18n.translate(
+        'xpack.apm.transactionActionMenu.showPodMetricsLinkLabel',
+        { defaultMessage: 'Show pod metrics' }
+      ),
+      condition: podId,
+      path: `/link-to/pod-detail/${podId}`,
+      query: infraMetricsQuery
+    },
+    {
+      icon: 'infraApp',
+      label: i18n.translate(
+        'xpack.apm.transactionActionMenu.showContainerMetricsLinkLabel',
+        { defaultMessage: 'Show container metrics' }
+      ),
+      condition: containerId,
+      path: `/link-to/container-detail/${containerId}`,
+      query: infraMetricsQuery
+    },
+    {
+      icon: 'infraApp',
+      label: i18n.translate(
+        'xpack.apm.transactionActionMenu.showHostMetricsLinkLabel',
+        { defaultMessage: 'Show host metrics' }
+      ),
+      condition: hostName,
+      path: `/link-to/host-detail/${hostName}`,
+      query: infraMetricsQuery
+    }
+  ].map(({ icon, label, condition, path, query }, index) =>
+    !!condition
+      ? {
+          icon,
+          key: `infra-link-${index}`,
+          child: (
+            <InfraLink path={path} query={query}>
+              {label}
+            </InfraLink>
+          )
+        }
+      : null
+  );
 
-    const items = [
-      ...this.getInfraActions(),
-      <EuiContextMenuItem icon="discoverApp" key="discover-transaction">
-        <EuiFlexGroup gutterSize="s">
-          <EuiFlexItem>
-            <DiscoverTransactionLink transaction={transaction}>
-              {i18n.translate(
-                'xpack.apm.transactionActionMenu.viewSampleDocumentLinkLabel',
-                {
-                  defaultMessage: 'View sample document'
-                }
-              )}
-            </DiscoverTransactionLink>
-          </EuiFlexItem>
-          <EuiFlexItem grow={false}>
-            <EuiIcon type="popout" />
-          </EuiFlexItem>
-        </EuiFlexGroup>
-      </EuiContextMenuItem>
-    ];
+  const uptimeLink =
+    transaction && transaction.url
+      ? url.format({
+          pathname: chrome.addBasePath('/app/uptime'),
+          hash: `/?${fromQuery(
+            pick(
+              {
+                dateRangeStart: urlParams.rangeFrom,
+                dateRangeEnd: urlParams.rangeTo,
+                search: `url.domain:${transaction.url.domain}`
+              },
+              (val: string) => !!val
+            )
+          )}`
+        })
+      : null;
```

Comment:
I think it's cleaner to add it to the items unconditionally, and then set the `condition` like:
```js
condition: transaction && transaction.url
```

Reply:
Will fix.

Reply:
Fixed.